### DISCUSSION TITLE
References for c128lib

### DISCUSSION THREAD

**Discussion by intoinside** [c128lib/.github#2] [Team Posts]
This thread is used to enumerate documentation used by c128lib and References project
- Commodore 128 ROM Disassembly https://klasek.at/c64/c128-rom-listing.html
- C128 User Callable KERNAL Routines https://cx16.dk/c128-kernal-routines/
- Mapping 128 [book] https://www.cubic.org/~doj/c64/mapping128.pdf
- Commodore 128 PROGRAMMER'S REFERENCE GUIDE [book] https://www.cubic.org/~doj/c64/c128ProgrammersReferenceGuide.pdf
- The Commodore 128 Subroutine Library  [book] https://ia800507.us.archive.org/28/items/The_Commodore_128_Subroutine_Library/The_Commodore_128_Subroutine_Library.pdf

**Comment by johnpalermo** [comment]
The best list.  

Can I recommend this one also.  

https://ia800507.us.archive.org/28/items/The_Commodore_128_Subroutine_Library/The_Commodore_128_Subroutine_Library.pdf

It has a number of C128 subroutines for various applications written in Basic, but can be used as a reference for building them in KickAssember.  I realize that many of these will intersect with existing routines already in the current c128lib, but others can be investigated to see if they would be a good fit for c128lib.  Maybe this can be a project.  I don't know.  Looking for feedback.

**Comment by intoinside** [comment]
This is an interesting book, could you give an example of what you mean? Just to make sure to understand what you are saying

**Comment by johnpalermo** [comment]
I have some examples for graphics, sound, file handling, joystick.  For example, if someone wants to develop a simple game and use some simple sounds for their game prototype, the book has sounds for things like a bomb dropping or a flying saucer.  The BASIC program would have to be disassembled into KickAssember code, and a macro or KickAss function could be added to the c128lib.  

Another example is a subroutine for simple joystick movement on the screen.  

Another example is string handling.  The book had various routines for handling strings that could be very useful for someone who wants to create a word game, or even a word processor.  

Another example is a simple function or macro to read/write and sequential file.  The book has some subroutines in BASIC for this.  Again, disassembly would be required to convert them to KickAssember code.

The book also has some nice functions for doing business type of calculations, which be handy for someone that wanted to write a business application for the C128.

Many of these would be built using the existing macros and functions in c128lib as it already includes a comprehensive chipset and memory library for graphics, sound, math, etc.

**Comment by intoinside** [comment]
Understood, and i think this is a great idea, these sounds like snippets, probably they can fit better in a new repository under c128lib, specific for snippet code?
This new repo should link common and/or chipset

**Comment by johnpalermo** [comment]
Sounds good.  Permission to create a new repo?

**Comment by intoinside** [comment]
I've updated permissions, now you should be able to create it, how do you call it? snippets?

Can you use projects to track activities? https://github.com/orgs/c128lib/projects/1

**Comment by johnpalermo** [comment]
To be honest, I’m not sure what to call it yet. I was originally thinking to make it as a library of high level macros and functions that users can call in their source code, kind of like a high level API, rather than copy/paste a snippet. The APIs would be wrappers to the common and chipset libs you’ve already created. I don’t know if api is a good term for it. I would prefer if I get some opinions on that as I’m just not sure myself. I guess that snippets would be a good name, but I’m wondering if it may confuse users, because normally snippets are generally understood to be part of IDE extensions that allow users to insert code templates, whereas my intent was to make the repository a library of macros and functions.

**Comment by intoinside** [comment]
As I read your comment, seems to me that you are describing a Framework. Framework could be the name of the repository while the macros and functions of the different topics could be divided into files.

**Comment by johnpalermo** [comment]
I agree. This sounds like a good term. I will use it. Thanks.
